common/MultiTarget/GenerateMultiTargetAwareProjectReferenceProps.ps1

```diff
@@ -0,0 +1,72 @@
+Param (    
+    [Parameter(HelpMessage = "The full path of the csproj to generated references to.", Mandatory = $true)] 
+    [string]$projectPath,
+
+    [Parameter(HelpMessage = "A path to a .props file where generated content should be saved to.", Mandatory = $true)] 
+    [string]$outputPath,
+
+    [Parameter(HelpMessage = "The path to the template used to generate the props file.")] 
+    [string]$templatePath = "$PSScriptRoot/MultiTargetAwareProjectReference.props.template",
+
+    [Parameter(HelpMessage = "The path to the props file that contains the default MultiTarget values.")] 
+    [string]$multiTargetDefaultPropsPath = "$PSScriptRoot/Defaults.props",
+
+    [Parameter(HelpMessage = "The placeholder text to replace when inserting the project file name into the template.")] 
+    [string]$projectFileNamePlaceholder = "[ProjectFileName]",
+
+    [Parameter(HelpMessage = "The placeholder text to replace when inserting the project path into the template.")] 
+    [string]$projectRootPlaceholder = "[ProjectRoot]"
+)
+
+$relativeProjectPath = (Resolve-Path -Path $projectPath);
+$templateContents = Get-Content -Path $templatePath;
+
+# Insert csproj file name.
+$csprojFileName = [System.IO.Path]::GetFileName($relativeProjectPath);
+$templateContents = $templateContents -replace [regex]::escape($projectFileNamePlaceholder), $csprojFileName;
+
+# Insert project directory
+$projectDirectory = [System.IO.Path]::GetDirectoryName($relativeProjectPath);
+$templateContents = $templateContents -replace [regex]::escape($projectRootPlaceholder), $projectDirectory;
+
+function LoadMultiTargetsFrom([string] $path) {
+    $fileContents = "";
+
+    # If file does not exist
+    if ($false -eq (Test-Path -Path $path -PathType Leaf)) {
+        # Load defaults
+        $fileContents = Get-Content $multiTargetDefaultPropsPath -ErrorAction Stop;
+    }
+    else {
+        # Load requested file
+        $fileContents = Get-Content $path -ErrorAction Stop;
+    }
+
+    # Parse file contents
+    $regex = Select-String -Pattern '<MultiTarget>(.+?)<\/MultiTarget>' -InputObject $fileContents;
+
+    if ($null -eq $regex -or $null -eq $regex.Matches -or $null -eq $regex.Matches.Groups -or $regex.Matches.Groups.Length -lt 2) {
+        Write-Error "Couldn't get MultiTarget property from $path";
+        exit(-1);
+    }
+
+    return $regex.Matches.Groups[1].Value;
+}
+
+# Load multitarget preferences for project
+$multiTargets = LoadMultiTargetsFrom("$projectDirectory\MultiTarget.props");
+$multiTargets = $multiTargets.Split(';');
+
+Write-Host "Generating project references for $([System.IO.Path]::GetFileNameWithoutExtension($csprojFileName)): $($multiTargets -Join ', ')"
+
+$templateContents = $templateContents -replace [regex]::escape("[CanTargetWasm]"), "'$($multiTargets.Contains("wasm").ToString().ToLower())'";
```

Comment:
Yeah, had troubles getting this working and had to resort to parsing in PowerShell. I'd like to do a bit of digging and come back to this later, after looking through the spec and source of MSBuild I know it can be done.

Reply:
Done in a353aed7fe41659bee39d0fa93edf555e7885bf6

![image](https://user-images.githubusercontent.com/9384894/208756430-46ab8bcc-57d8-4b03-a3ae-6a64176ab99e.png)